services/management/ia/templates_service.go

```diff
@@ -464,14 +464,15 @@ func (s *TemplatesService) UpdateTemplate(ctx context.Context, req *iav1beta1.Up
 		return nil, status.Error(codes.InvalidArgument, "Request should contain exactly one rule template.")
 	}
 
-	for _, t := range templates {
-		if err = validateUserTemplate(&t); err != nil { //nolint:gosec
-			return nil, status.Errorf(codes.InvalidArgument, "%s.", err)
-		}
+	tmpl := templates[0]
+
+	if err = validateUserTemplate(&tmpl); err != nil { //nolint:gosec
+		return nil, status.Errorf(codes.InvalidArgument, "%s.", err)
```

Comment:
Probably you don't need to expose the error to the client, but just log it.

```suggestion
		s.l.Errorf("failed to validate alert template from request: +%v", err)
		return nil, status.Error(codes.InvalidArgument, "Failed to validate alert template.")
```

Reply:
Are you sure? For me as user, is good to know that "template name should not start with 'pmm_' or 'saas_' prefix"

Reply:
Ok then.